File: src/main/java/net/benwoodworth/fastcraft/core/ConsoleLogger.java
package net.benwoodworth.fastcraft.core;

import org.jetbrains.annotations.NotNull;

public interface ConsoleLogger {

    /**
     * Log an error to the console.
     *
     * @param message The error message to log.
     */
    void error(@NotNull String message);

    /**
     * Log info to the console.
     *
     * @param message The info to log.
     */
    void info(@NotNull String message);

    /**
     * Log a warning to the console.
     *
     * @param message The warning to log.
     */
    void warn(@NotNull String message);

    /**
     * Log a debug message to the console, only if debug output is enabled.
     *
     * @param message The debug message to log.
     */
    void debug(@NotNull String message);
}

File: src/main/kotlin/net/benwoodworth/mc/fastcraftplus/core/ConsoleLogger.kt
package net.benwoodworth.mc.fastcraftplus.core

interface ConsoleLogger {

    /**
     * Log an error to the console.
     *
     * @param message The error message to log.
     */
    fun error(message: String)

    /**
     * Log info to the console.
     *
     * @param message The info to log.
     */
    fun info(message: String)

    /**
     * Log a warning to the console.
     *
     * @param message The warning to log.
     */
    fun warn(message: String)

    /**
     * Log a debug message to the console, only if debug output is enabled.
     *
     * @param message The debug message to log.
     */
    fun debug(message: String)
}
